example/src/app/a-bit-more-complex-component/a-bit-more-complex.component.ts

```diff
@@ -0,0 +1,28 @@
+import { Component, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'app-a-bit-more-complex',
+  template: `
+    <div class="kek">
+      diary works!
+      <div *ngIf="someVar" class="inner-class">
+        <div *ngIf="anotherVar" class="inner-hidden"></div>
+        <div>another case</div>
+        <span>some html</span>
+      </div>
+      <div>one more case case</div>
+      <div><app-child-component kek="someVar"></app-child-component></div>
+    </div>
+  `,
+  styles: [`
+    .kek { color: red }
+  `]
```

Comment:
I still think we do not need the styles at all. No one ever looks at the components and they just add noise.

The browser/jest does not complain if there is not an actual class with the name `kek` loaded.

Reply:
Actually.... We need styles or style urls in case of that fact. If styles are provided to the component they must be encapsulated (by default). I don't know why, but even that fact that I specify style (inline in component selector) or styleUrls - these attributes nghost/ngcontent are not created for some reasons. Looks like there is come code that removes them....

My initial code with style was about that. I don't understand why nghost/ngcontent attributes are not in snapshots now.

Reply:
Will we handle nghost/ngcontent in this PR?

Or do you understand why this attributes are not in result snapshot?

Reply:
Good point.
I have no clue why they disappeared. Let's keep investigating a bit more.

Reply:
I know why. We strip the styles in the preset, as we do not care about them in jest tests.

But there are several ways to avoid the stripping. This one should work (though I haven't tested it):
```ts
 @Component({
  selector: 'app-medium-complex',
  template: `...`,
  ['styles']: [`
    .kek { color: red }
  `]
})
```

Reply:
Ok. In that case... should I handle nghost/ngcontent if you trim styles anyway?

By the way.... Where is that code what removes styles from the components?

Reply:
Yeah, let's handle them just in case.

Reply:
Also, what about ng-version? I don't have code for trimming version at the moment. But still somebody got ng-version (this parameter is dynamic and snapshots mustn't relay on such parameter without doubts). @listepo-alterpost had that problem for example.

Reply:
I think `ng-version` is only added to the root app, I *guess* it is only rendered when Angular is bootstrapped with a root module/component, not using `TestBed` (might be wrong about it).
I do not see why someone would serialize a whole app using jest, so I do not think this is required, but go ahead if you like to add it.

Reply:
> 

Not really. Change snapshot test in simple component to `expect(fixture.nativeElement).toMatchSnapshot();`. You will get this stuff:

![image](https://user-images.githubusercontent.com/5622526/56738282-f4a78f80-6774-11e9-9856-769e320c33d2.png)

SimpleComponent is not root component, but still it gets ng-version attribute. I think that in in test environment.

Also, here is full dom snapshot starting from body tag:

```
// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleComponent snapshots 1`] = `
<body>
  <div
    id="root1"
    ng-version="7.2.8"
  >
    <p>
       simple works!

    </p>
  </div>
</body>
`;
```

I tried `expect(document.querySelector('body')).toMatchSnapshot();` this expect.

So, if anybody will snapshot full body (which is strange) or anybody fill snapshot not fixture, but fixture.nativeElement (more real, as I think) he will get this problem.

> Edit: we can still add it later if it gets requested.

It might be breaking change after we initially merge this to the project. So better to implement this now, if it is possible.

So now: 

- I add example with styles and `ngcontent`/`nghost` properties and I will add code to new serializer to remove them (despite of `jest-preset-angular` removes styles anyway)
- I remove `ng-version` attributes too
- I need fix docs (or don't?). We need some discussion I think. If I understand correcly, you don't want this serializer by default. So I have to add section how to remove angular specific attributes by adding new serializer. Can you clarify how this will work in general, so I can describe it in docs?

Reply:
Ok cool! I was not aware how it can be reproduced.

> I add example with styles and ngcontent/nghost properties
> I remove ng-version attributes too

Yeah, we need tests for this. But maybe we can reuse one of the components already created.

> I need fix docs

The serializer is currently configured in 
https://github.com/thymikee/jest-preset-angular/tree/master/jest-preset.js. We can add a commented out line `// 'jest-preset-angular/AngularSpecificAttributesRemover.js',`, but leave it in there.
We should also put it in the exposed configuration in `README.md` and add some lines about the new serializer being available.
As soon as `jest-preset-angular` will release v8, I would like to propose to use this serializer by default, then we can document it as breaking change as well.
This way we can already publish it and users can already try it out.
The old serializer will still be available for users that prefer the old serialization. 

So these steps are required for the docs:
* Add short section in `README.md`, a bullet point in https://github.com/thymikee/jest-preset-angular#brief-explanation-of-config and the addition of the commented out serializer is sufficient
* A short description in `CHANGELOG.md` including a quick migration guide

Reply:
KK, I need some time. Will be done todays evening.

Reply:
Hm, what section in readme describes how to rewrite default `jest-preset.js` configuration?

I do not understand how end-user can rewrite `snapshotSerializer` settings. Could you explain this please?

We need to decide what we want in example app?
- snapshots with angular-specific attributes (since new serializer is commented by default)
- clearer snapshots without such attributes

How to rewritet this setting ? o_O